mne/preprocessing/maxwell.py

```diff
@@ -287,142 +304,384 @@ def maxwell_filter(raw, origin='auto', int_order=8, ext_order=3,
+        avg_trans = np.vstack([
+            np.hstack([quat_to_rot(avg_trans[:3]),
+                       avg_trans[3:][:, np.newaxis]]),
+            [[0., 0., 0., 1.]]])
+    return trans, rel_starts, rel_stops, quats, avg_trans
+
+
+def _do_tSSS(clean_data, orig_in_data, resid, st_correlation,
+             n_positions, t_str):
+    """Compute and apply SSP-like projection vectors based on min corr"""
+    np.asarray_chkfinite(resid)
+    t_proj = _overlap_projector(orig_in_data, resid, st_correlation)
+    # Apply projector according to Eq. 12 in [2]_
+    msg = ('        Projecting % 2d intersecting tSSS components '
+           'for %s' % (t_proj.shape[1], t_str))
+    if n_positions > 1:
+        msg += ' (across % 2d positions)' % n_positions
+    logger.info(msg)
+    clean_data -= np.dot(np.dot(clean_data, t_proj), t_proj.T)
+
+
+def _copy_preload_add_channels(raw, add_channels):
+    """Helper to load data for processing and (maybe) add cHPI pos channels"""
+    raw = raw.copy()
+    if add_channels:
+        kinds = [FIFF.FIFFV_QUAT_1, FIFF.FIFFV_QUAT_2, FIFF.FIFFV_QUAT_3,
+                 FIFF.FIFFV_QUAT_4, FIFF.FIFFV_QUAT_5, FIFF.FIFFV_QUAT_6,
+                 FIFF.FIFFV_HPI_G, FIFF.FIFFV_HPI_ERR, FIFF.FIFFV_HPI_MOV]
+        out_shape = (len(raw.ch_names) + len(kinds), len(raw.times))
+        out_data = np.zeros(out_shape, np.float64)
+        msg = '    Appending head position result channels and '
+        if raw.preload:
+            logger.info(msg + 'copying original raw data')
+            out_data[:len(raw.ch_names)] = raw._data
+            raw._data = out_data
+        else:
+            logger.info(msg + 'loading raw data from disk')
+            raw._preload_data(out_data[:len(raw.ch_names)], verbose=False)
+            raw._data = out_data
+        assert raw.preload is True
+        off = len(raw.ch_names)
+        chpi_chs = [
+            dict(ch_name='CHPI%03d' % (ii + 1), logno=ii + 1,
+                 scanno=off + ii + 1, unit_mul=-1, range=1., unit=-1,
+                 kind=kinds[ii], coord_frame=FIFF.FIFFV_COORD_UNKNOWN,
+                 cal=1. / 10000., coil_type=FIFF.FWD_COIL_UNKNOWN)
+            for ii in range(len(kinds))]
+        raw.info['chs'].extend(chpi_chs)
+        raw.info['nchan'] += len(chpi_chs)
+        raw.info['ch_names'] += [c['ch_name'] for c in chpi_chs]
+        raw.info._check_consistency()
+        assert raw._data.shape == (raw.info['nchan'], len(raw.times))
+        # Return the pos picks
+        pos_picks = np.arange(len(raw.ch_names) - len(chpi_chs),
+                              len(raw.ch_names))
+        return raw, pos_picks
+    else:
+        if not raw.preload:
+            logger.info('    Loading raw data from disk')
+            raw.load_data(verbose=False)
+        else:
+            logger.info('    Using loaded raw data')
+        return raw, np.array([], int)
+
+
+def _check_pos(pos, head_frame, raw, st_fixed):
+    """Check for a valid pos array"""
```

Comment:
Do you want to add something to the docstring about how it also returns the given info in a new format?